Sources/UI/Controllers/Wizard/Wizard.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  Wizard.swift
+//  PovioKit
+//
+//  Created by Toni Kocjan on 29/09/2021.
+//  Copyright © 2021 Povio Labs. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol WizardStep: UIViewController {
+}
+
+public class WizardDataSource: NSObject {
+  public typealias LazyStep = () -> WizardStep
+  
+  var currentStepIndex = -1
+  var currentStep: WizardStep?
+  var steps: [LazyStep]
+  
+  init(steps: [LazyStep]) {
+    self.steps = steps
+  }
+}
+
+public extension WizardDataSource {
+  func nextStep() -> WizardStep? {
+    currentStepIndex += 1
+    guard let thunk = steps[safe: currentStepIndex] else {
+      currentStepIndex -= 1
+      return nil
+    }
+    let step = thunk()
+    self.currentStep = step
+    return step
+  }
+  
+  func previousStep() -> WizardStep? {
+    currentStepIndex -= 1
+    guard let thunk = steps[safe: currentStepIndex] else {
+      currentStepIndex += 1
+      return nil
+    }
+    let step = thunk()
+    self.currentStep = step
+    return step
+  }
+}
+
+/// An automatically resizable dialog superclass.
+///
+/// This view controller enables you to add and transition
+/// between view controllers inside a dialog.
+/// The UI, presentation, and layout are completely decoupled,
+/// allowing for maximum specialization.
+///
+open class Wizard<ContentView: WizardContentView>: UIViewController {
+  public typealias DataSource = WizardDataSource
+  public typealias LazyStep = DataSource.LazyStep
+  
+  public let contentView: ContentView
+  let dataSource: DataSource
+  
+  public init(
+    steps: [LazyStep],
+    contentView: ContentView
+  ) {
```

Comment:
Whose `contentView` is this and why do we need to inject it? Maybe an example/documentation on how this is used would be great 😄